### PR TITLE
kbfsgit: no lazy creates for team or single-user private TLFs

### PR DESCRIPTION
**Pull request by strib**
Issue: KBFS-2480

### CODE REVIEW

**Comment by strib** [comment]
> So the only way to create a new repo now is through GUI or keybase git right?

Yes, for your personal private repo and for team TLFs.

**Comment by songgao** [comment]
Got you. Thanks! SGTM. (Yeah meant to ask for personal private repo and team TLFs)

File: kbfsgit/runner.go [hunk 1]
```diff
@@ -251,8 +251,14 @@ func (r *runner) initRepoIfNeeded(ctx context.Context, forCmd string) (
 		}()
 	}
 
-	fs, _, err = libgit.GetOrCreateRepoAndID(
-		ctx, r.config, r.h, r.repo, r.uniqID)
+	// Only allow lazy creates for public and multi-user TLFs.
```
Comment (jzila) [inline kbfsgit/runner.go]: Is that because we don't provide a UI for these?

File: kbfsgit/runner.go [hunk 2]
```diff
@@ -251,8 +251,14 @@ func (r *runner) initRepoIfNeeded(ctx context.Context, forCmd string) (
...
+	if r.h.Type() == tlf.Public || len(r.h.ResolvedWriters()) > 1 {
```
Comment (jzila) [inline kbfsgit/runner.go]: What about unresolved writers? I should be able to make a git repository with `jzila,foo@twitter`.

File: libgit/repo.go [hunk 3]
```diff
@@ -238,7 +271,7 @@ func getOrCreateRepoAndID(
 		return nil, NullID, err
 	}
 
-	if createOnly {
+	if op == createOnly {
```
Comment (jzila) [inline libgit/repo.go]: How are we locking this?

File: kbfsgit/runner.go (hunk 1 above)
Comment (strib) [reply kbfsgit/runner.go]: Yeah, there would be no other way to create public or mutli-user repos otherwise.

File: kbfsgit/runner.go (hunk 2 above)
Comment (strib) [reply kbfsgit/runner.go]: Good point, will fix.

File: libgit/repo.go (hunk 3 above)
Comment (strib) [reply libgit/repo.go]: I don't think what "this" means here.  If we get to this line of code, we know for sure the repo existed at the time the request started.  That's enough to fail the request.  If this is being executed concurrently with a delete of the same repo, oh well, the user will just have to retry.  Does that make sense?

Comment (jzila) [reply libgit/repo.go]: Yep, thanks.

**Pull request merged**